### PR TITLE
NETWORK_PERMISSION and NETWORK_AUTH replaced by strong enums

### LINKED ISSUE #12420: Refactor NETWORK_PERMISSION to use strong enum

### LINKED ISSUE #12421: Refactor NETWORK_AUTH to use strong enum

### PR DESCRIPTION
**Pull request by ju-pinheiro**
Close #12420 
Close #12421

### CODE REVIEW

**Comment by mwnciau** [comment]
We'd probably want to include those other functions as part of the refactor. Just having a quick look, you could remove the majority of the static_casts by moving the cast into the ToggleActionPermission method instead.

I see you've fixed the clang formatting already. We usually use git rebase / fixup to merge those changes into the original commit. This keeps the commit history clean.

Other than that, looks good!

**Comment by ju-pinheiro** [comment]
Hi Simon, moving the cast into the ToggleActionPermission method does seem a better idea, at least it would avoid duplicates. I'll review it, thanks! :)
And also thanks for the git tip to merge the fix commit!

File: src/openrct2-ui/windows/Player.cpp [hunk 1]
```diff
@@ -468,7 +468,8 @@ void window_player_overview_invalidate(rct_window* w)
     }
 
     // Only enable kick button for other players
-    const bool canKick = network_can_perform_action(network_get_current_player_group_index(), NETWORK_PERMISSION_KICK_PLAYER);
+    const bool canKick = network_can_perform_action(
+        network_get_current_player_group_index(), static_cast<uint32_t>(NetworkPermission::KickPlayer));
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Player.cpp:472]: Use the new type for network_can_perform_action

**Review by mwnciau — COMMENTED** [review]
Looks a lot better now.

@ju-pinheiro you should mark duncan's comments above as resolved if you've addressed them.

**Comment by tupaschoal** [comment]
> @ju-pinheiro you should mark duncan's comments above as resolved if you've addressed them.

Please always do this

File: src/openrct2/network/NetworkGroup.h [hunk 2]
```diff
@@ -16,6 +16,8 @@
 #include <jansson.h>
 #include <string>
 
+enum class NetworkPermission : uint32_t;
```
[the remark this answers is not in the document]
Comment (tupaschoal) [reply src/openrct2/network/NetworkGroup.h:19]: While I agree `NetworkAction.h` is not a problem and could be included here, I don't mind erring on the side of having more forward declarations.

**Review by duncanspumpkin — APPROVED** [review]
This is really showing some bad naming in the network functions but not for this pr to address.

**Pull request merged**